azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/amqp/BaseLinkHandler.java

```diff
@@ -38,39 +37,35 @@ public void onLinkLocalClose(Event event) {
     public void onLinkRemoteClose(Event event) {
         final Link link = event.getLink();
 
-        if (link.getLocalState() != EndpointState.CLOSED) {
-            link.close();
-        }
-
         if (link != null) {
```

Comment:
In the onLinkXXX event handlers - link cannot be null. I introduced these null checks - when we had a bug - due to which reactor stack ran in multi-threaded fashion - and later missed to remove them. 
lets get rid of the if (link != null) code.

Reply:
Ok. let us remove null check. But having this in place is no harm.